fact/path/__init__.py

```diff
@@ -0,0 +1,114 @@
+import os.path
+import re
+
+__all__ = [
+    'parse'
+    'run2tree_path',
+    'TemplateToPath',
+    'TreePath',
+]
+
+
+class TemplateToPath:
+    '''Turn a template like '/fac/raw/{Y}/{M}/{D}/{N}_{R}.fits.fz' into a path.
+    '''
+
+    def __init__(self, template):
+        self.template = template
+
+    def __call__(self, night, run=None, **kwargs):
+        d = dict(**kwargs)
+
+        if hasattr(night, 'fNight') and hasattr(night, 'fRunID'):
+            night, run = night.fNight, night.fRunID
+
+        night = str(night)
+        if run is not None:
+            d['R'] = '{:03d}'.format(int(run))
+
+        d['N'] = night
+        d['Y'] = night[0:4]
+        d['M'] = night[4:6]
+        d['D'] = night[6:8]
+        return self.template.format(**d)
+
+
+def run2tree_path(base_dir, suffix, night, run=None):
+    '''Make a tree_path from a run
+
+    base_dir:  eg. '/fact/raw' or '/fact/aux'
+    suffix:    eg. '.fits.fz' or '.log' or '.AUX_FOO.fits'
+    night:     eg. 20160101 or '20160101' (int or string accepted)
+    run:       eg. 11 or '011' or None (int, string or None accepted)
+
+    output:
+    eg. '/fact/raw/2016/01/01/20160101_011.fits.fz' or
+    '/fact/raw/2016/01/01/20160101.log'
+
+    '''
+    if run is not None:
+        base_name = '{N}_{R}'
+    else:
+        base_name = '{N}'
+
+    template = os.path.join(
+            base_dir,
+            '{Y}',
+            '{M}',
+            '{D}',
+            base_name + suffix)
+    return TemplateToPath(template)(night, run)
+
+
+class TreePath:
```

Comment:
Same thing as above, it even defines `__call__`.

Reply:
right